tt_metal/detail/tt_metal.hpp

```diff
@@ -384,7 +381,7 @@ namespace tt::tt_metal{
                 soc_d.get_physical_ethernet_cores(),
                 soc_d.grid_size,
                 harvested_rows,
-                {device->worker_core_from_logical_core(consumer_logical_core)});
+                device->worker_core_from_logical_core(enqueue_program_dispatch_core));
```

Comment:
do we only supply one consumer core because currently only one cq (the first cq) will be servicing enqueue program commands?

Reply:
In one way, yes. The other problem is that I could not get @pgkeller 's dev message struct extended without getting hangs (needs to be extended in order for us to specify the dispatch core at runtime). So for now, I assert only one cq can be the program dispatcher.

Reply:
ah makes sense, perhaps open an issue for that to track